### PR TITLE
Evalute support task_type evaluate

### PR DESCRIPTION
**Pull request by singankit**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ai/azure-ai-generative/azure/ai/generative/evaluate/_evaluate.py [hunk 1]
```diff
@@ -208,7 +209,11 @@ def _evaluate(
 
         def _get_instance_table():
```
Comment (luigiw) [inline sdk/ai/azure-ai-generative/azure/ai/generative/evaluate/_evaluate.py:210]: Nit pick: why is this an inline function?

File: sdk/ai/azure-ai-generative/azure/ai/generative/evaluate/_metric_handler.py [hunk 2]
```diff
@@ -74,13 +74,19 @@ def _get_data_for_metrics(self):
...
+        from azureml.metrics import compute_metrics, constants
+        from ._constants import CHAT
 
         metrics_calculation_data = self._get_data_for_metrics()
 
+        metrics = None
+        if self.metrics is None and self.task_type == CHAT:
+            metrics = []
+        else:
+            metrics = self.metrics
```
Comment (luigiw) [inline sdk/ai/azure-ai-generative/azure/ai/generative/evaluate/_metric_handler.py:86]:
```suggestion
        metrics = self.metrics if self.task_type != CHAT else [] 
```

**Review by luigiw — APPROVED** [review]

File: sdk/ai/azure-ai-generative/azure/ai/generative/evaluate/_evaluate.py (hunk 1 above)
Comment (singankit) [reply sdk/ai/azure-ai-generative/azure/ai/generative/evaluate/_evaluate.py:210]: It started as an inline way to convert data to jsonl that is needed in artifacts produced. Long term plan is to break it down into different eval scenarios like qa, chat.

**Pull request merged**